app/packages/state/src/hooks/similaritySearch/useSimilarity.test.tsx

```diff
@@ -0,0 +1,133 @@
+}) => {
+  return (
+    <RecoilRoot
+      initializeState={({ set }) => {
+        set(fos.dataset, TEST_DS);
+        set(fos.selectedSamples, new Set(["1", "2"]));
+      }}
+    >
+      {children}
+    </RecoilRoot>
+  );
+};
+
+const RootWithActiveImageSort: React.FC<React.PropsWithChildren<{}>> = ({
+  children,
+}) => {
+  return (
+    <RecoilRoot
+      initializeState={({ set }) => {
+        set(fos.dataset, TEST_DS);
+        set(fos.similarityParameters, {
+          brainKey: "test-brain-key",
+          k: 10,
+          queryIds: ["1"],
+        });
+      }}
+    >
+      {children}
+    </RecoilRoot>
+  );
+};
+
+const RootWithSelectedLabelModalView: React.FC<React.PropsWithChildren<{}>> = ({
+  children,
+}) => {
+  return (
+    <RecoilRoot
+      initializeState={({ set }) => {
+        set(fos.dataset, TEST_DS);
+        set(fos.selectedLabels, {
+          "1": {
+            sampleId: "1",
+            field: "test-field",
+          },
+        });
+        set(fos.currentModalSample, { id: "1", index: 1234 });
+      }}
+    >
+      {children}
+    </RecoilRoot>
+  );
+};
+
+describe("similarity search helper text and icon are correct", () => {
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  test("Default state show text similarity search", async () => {
+    const { result } = renderHook(
+      () => fos.useSimilarityType({ isImageSearch: true }),
+      { wrapper: Root }
+    );
+    expect(result.current.text).toBe("Sort by text similarity");
+    expect(result.current.showImageSimilarityIcon).toBe(false);
+  });
+
+  test("when samples are selected, image similiarty search", async () => {
+    const { result } = renderHook(
+      // isImageSearch is false/true does not impact this test scenario
+      () => fos.useSimilarityType({ isImageSearch: false }),
+      { wrapper: RootWithSelectedSamples }
+    );
+    expect(result.current.text).toBe("Search by image similarity");
+    expect(result.current.showImageSimilarityIcon).toBe(true);
+  });
+
+  test("when an image similarity is done and extended stage has sorting setting, should show image icon", () => {
+    const { result } = renderHook(
+      () => fos.useSimilarityType({ isImageSearch: true }),
+      { wrapper: RootWithActiveImageSort }
+    );
+    expect(result.current.text).toBe("Search by image similarity");
+    expect(result.current.showImageSimilarityIcon).toBe(true);
+  });
+
+  test("when an text similarity is done and extended stage has sorting setting, should show text icon", () => {
+    const { result } = renderHook(
+      () => fos.useSimilarityType({ isImageSearch: false }),
+      { wrapper: RootWithActiveImageSort }
+    );
+    expect(result.current.text).toBe("Sort by text similarity");
+    expect(result.current.showImageSimilarityIcon).toBe(false);
+  });
+
+  test("when labels are selected in sample modal view, should show image icon", () => {
+    // isImageSearch is false/true does not impact this test scenario
```

Comment:
you could make the `isImageSearch` property optional with `false` value so you don't have to pass it here and no comment needed